sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client.py

```diff
@@ -104,6 +104,7 @@ def __init__(
             self._hosts = {LocationMode.PRIMARY: primary_hostname, LocationMode.SECONDARY: secondary_hostname}
 
         self.require_encryption = kwargs.get("require_encryption", False)
+        self.encryption_version = kwargs.get("encryption_version", "1.0")
```

Comment:
This feature is completely undocumented for Queue and I think we may be leaving it like that. I think we will be adding a warning if the user configures encryption but does not set this  to 2.0. Still sort of an open dicussion on that though so will leave it out for now.

@annatisch curious on your thoughts of adding a warning here if the user adds a key_encryption_key (enables encryption) but does not set this version to 2.0? Sort of a deprecation warning for 1.0. This may be something we want to do after preview? There is an open mail thread on this as well.

Reply:
Yes - we should add a warning if customers are passing in the other encryption parameters but not setting the version to 2.0.

Reply:
I've added a warning.